packages/mattermost-redux/src/actions/users.ts

```diff
@@ -84,22 +88,19 @@ export function loadMeREST(): ActionFunc {
                 dispatch(getMyTeamMembers()),
             ]);
 
+            userId = actionResponse?.[2]?.data?.id ?? '';
+            userRolesAsString = actionResponse?.[2]?.data?.roles ?? ''; // for rest api, roles are returned as (role1<space>role2<space>role3)
+
             const isCollapsedThreads = isCollapsedThreadsEnabled(state);
             await dispatch(getMyTeamUnreads(isCollapsedThreads));
         } catch (error) {
-            dispatch(logError(error));
-            return {error};
+            dispatch(logError(error as ServerError));
+            return {error: error as ServerError};
         }
 
-        const {currentUserId} = state.entities.users;
```

Comment:
I see what was happening before, so we don't actually need to pull this data from the result of the `Promise.all`. The problem that this had before is that we're still using an old version of `state` from before those requests were made, so it's missing the data that was loaded from after it was called on line 71. If we just call `getState` again to get an updated copy of the state, it should work the way it was written before

Reply:
yes changed